queries.sql

```diff
@@ -7,3 +7,60 @@ SELECT name, escape_attempts from animals WHERE weight_kg > 10.5;
 SELECT * FROM animals WHERE neutered=true;
 SELECT * FROM animals WHERE name<>'Gabumon';
 SELECT * FROM animals WHERE weight_kg BETWEEN 10.4 AND 17.3;
+
+-- Task 2 queries.
+-- No 1.
+BEGIN;
+UPDATE animals
+SET species = 'unspecified';
+SELECT * FROM animals;
+ROLLBACK;
+
+-- No 2.
+BEGIN;
+
+UPDATE animals 
+SET species = 'digimon'
+WHERE name like '%mon';
+
+UPDATE animals 
+SET species = 'pokemon'
+WHERE species = '';
+
+COMMIT;
+SELECT * FROM animals;
+
+--No 3.
+BEGIN;
+DELETE FROM animals;
+ROLLBACK;
+SELECT * FROM animals;
+
+--No 4.
+BEGIN;
+DELETE FROM animals WHERE date_of_birth > '2022-01-01';
+SELECT * FROM animals;
+SAVEPOINT checkpoint1;
+UPDATE animals SET weight_kg=weight_kg*-1;
+SELECT * FROM animals;
+ROLLBACK TO checkpoint1;
+UPDATE animals SET weight_kg=weight_kg*-1 where weight_kg<0;
+SELECT * FROM animals;
+COMMIT;
+
+--No 5
+SELECT COUNT(*) AS total FROM animals;
+
+SELECT COUNT(*) AS coolAnimals FROM animals
+WHERE escape_attempts=0;
+
+SELECT AVG(weight_kg) AS meanWeight FROM animals;
+
+SELECT neutered, SUM(escape_attempts) AS most_escape_attempts FROM animals GROUP BY neutered;
+
+SELECT species, MIN(weight_kg) AS min_weight, MAX(weight_kg) AS max_weight FROM animals GROUP BY species;
+
+SELECT species, AVG(escape_attempts) AS mean_escape_attempts FROM animals WHERE date_of_birth BETWEEN '1990-01-01' AND '2000-12-31' GROUP BY species ;
+
+
```

Comment:
- [ ] I think it would be better if you update this query to match the answer to the asked question. This is because the current excludes animals born on` '1990-01-01' `AND` '2000-12-31'`. To fix this you can use `>=` and `<=` instead of `BETWEEN`. You can read **[this article](https://stackoverflow.com/questions/1630239/sql-between-vs-and)** for more explanation about `BETWEEN` vs  `>=` and `<=`.